### PR TITLE
Feature/43328 ps report transformation

### PR DESCRIPTION
**Pull request by jon-shipley**
Create the Psychometric report for a single pupil

Input: service bus queue message
Output: service bus queue message

### CODE REVIEW

File: func-consumption/disable-functions.env [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
-AzureWebJobs.pupil-auth.Disabled=false
-AzureWebJobs.pupil-feedback.Disabled=false
-AzureWebJobs.pupil-login.Disabled=false
-AzureWebJobs.pupil-prefs.Disabled=false
-AzureWebJobs.school-pin-generator.Disabled=false
-AzureWebJobs.school-pin-http-service.Disabled=false
-AzureWebJobs.school-results-cache-determiner.Disabled=false
-AzureWebJobs.school-results-cache.Disabled=false
+AzureWebJobs.ps-report-transformed.Disabled=false
+AzureWebJobs.pupil-auth.Disabled=ture
```
Comment (GuyHarwood) [inline func-consumption/disable-functions.env:11]: misspelling

File: tslib/src/functions/ps-report-transformer/mocks/answers.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import moment from 'moment'
+import { Answer } from '../../../functions-throttled/ps-report-2-pupil-data/models'
```
Comment (GuyHarwood) [inline tslib/src/functions/ps-report-transformer/mocks/answers.ts:2]: might be better to pull this back into the root models folder?

File: tslib/src/functions/ps-report-transformer/report-line.class.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    DOB: '',
+    Gender: '',
+    PupilID: '',
+    Forename: '',
+    Surname: '',
+    ReasonNotTakingCheck: null
+  }
+
+  constructor (
+    answers: AnswersOrNull,
```
Comment (GuyHarwood) [inline tslib/src/functions/ps-report-transformer/report-line.class.ts:31]: if i recall correctly there is a shorthand way of doing constructor params with backing fields.
not always ideal, but worth knowing

**Review by GuyHarwood — COMMENTED** [review]
LGSF

File: tslib/src/functions/ps-report-transformer/report-line.class.ts (hunk 3 above)
Comment (jon-shipley) [reply tslib/src/functions/ps-report-transformer/report-line.class.ts:31]: Yes, I am aware.  But here we are using getters and setters so the `answers` param is being set to the `_answers` property.

**Review by GuyHarwood — COMMENTED** [review]
LGSF

**Review by GuyHarwood — APPROVED** [review]
LGTM 🏆

File: tslib/src/functions/ps-report-transformer/mocks/answers.ts (hunk 2 above)
Comment (GuyHarwood) [reply tslib/src/functions/ps-report-transformer/mocks/answers.ts:2]: did you look into this @jon-shipley ?

Comment (jon-shipley) [reply tslib/src/functions/ps-report-transformer/mocks/answers.ts:2]: I did an initial investigation, but more time would be needed to determine the right fit here.

**Pull request merged**